### DISCUSSION TITLE
Reservations

### DISCUSSION THREAD

**Discussion by craditsch** [Azure/ipam#22] [General]
Are reservations currently not yet implemented or do I oversee something?

**Comment by DCMattyG** [comment]
Hello @craditsch, thanks so much for reaching out!

Yes, Reservations are indeed implemented, albeit we need to add some better documentation on how-to for those. You can create a reservation in a Block by issuing a POST to:

https://&lt;IPAM ADDRESS&gt;/api/spaces/&lt;SPACE&gt;/blocks/&lt;BLOCK&gt;/reservations

The body for the request is as follows:

```
{
    size: <SUBNET BITS>
}
```

Example:

```
{
    size: 24
}
```

This will return an object that contains the next available vNET from the Block of the given size, plus a TAG you can place on the newly created vNET in Azure that IPAM will find and automatically associate the new vNET with the target Block it was reserved from (thus completing the reservation process). You can find the details around this API and it's response in the Swagger UI, which you can find here:

https://&lt;IPAM ADDRESS&gt;/api/docs

I hope that helps, and please reach out with any other questions/comments!

**Reply by craditsch** [reply]
Thank you @DCMattyG for the fast response, this reservation process is awesome and can really help us to streamline our AppZone automation.

**Comment by jonprattC78** [comment]
@DCMattyG - Is there any concept of 'time to expire' for a Reservation? 
Or would unclaimed Reservations build up in the database, requiring manual intervention at a point of eventual address space issues...

I will admit that I don't see a situation where this would be a thing - but I have been asked by internal mgmt., and whilst I have told them 'reservations would build up if not claimed' I have been asked to clarify this with you.

Thanks :)

**Reply by DCMattyG** [reply]
Hi @jonprattC78, that's a great question!

As of right now, there is no expiration. The idea is that the Network Operations team (or whomever manages IPAM) would clear out any reservations that were either A) created in error B) no longer required, or C) were stale for a period of time.

We've certainly talked about adding in some automatic expirations, so if you have an ideal of what that would/should look like in an ideal world, just let me know and I can work on getting something implemented.
